test/db_repo_query_test.go

```diff
@@ -0,0 +1,149 @@
+//+build integration
+
+package test_test
+
+import (
+	"context"
+	"github.com/applike/gosoline/pkg/cfg"
+	"github.com/applike/gosoline/pkg/db-repo"
+	"github.com/applike/gosoline/pkg/mdl"
+	"github.com/applike/gosoline/pkg/mon"
+	"github.com/applike/gosoline/pkg/test"
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+var TestModelMetadata = db_repo.Metadata{
+	ModelId: mdl.ModelId{
+		Application: "application",
+		Name:        "testModel",
+	},
+	TableName:  "test_models",
+	PrimaryKey: "test_models.id",
+	Mappings: db_repo.FieldMappings{
+		"testModel.id":   db_repo.NewFieldMapping("test_models.id"),
+		"testModel.name": db_repo.NewFieldMapping("test_models.name"),
+	},
+}
+
+type TestModel struct {
+	db_repo.Model
+	Name *string
+}
+
+type DbRepoQueryTestSuite struct {
+	suite.Suite
+	logger mon.Logger
+	config cfg.Config
+	mocks  *test.Mocks
+	repo   db_repo.Repository
+}
+
+func TestDbRepoQueryTestSuite(t *testing.T) {
+	suite.Run(t, new(DbRepoQueryTestSuite))
+}
+
+func (s *DbRepoQueryTestSuite) SetupSuite() {
+	setup(s.T())
+	mocks, err := test.Boot("test_configs/config.mysql.test.yml")
+
+	if !s.NoError(err) {
+		return
+	}
+
+	s.mocks = mocks
+
+	config := cfg.New()
+	_ = config.Option(
+		cfg.WithConfigFile("test_configs/config.mysql.test.yml", "yml"),
+		cfg.WithConfigFile("test_configs/config.db_repo_query.test.yml", "yml"),
+		cfg.WithConfigMap(map[string]interface{}{
+			"db.default.uri.port": s.mocks.ProvideMysqlPort("mysql"),
+			"db.default.uri.host": s.mocks.ProvideMysqlHost("mysql"),
+		}),
+	)
+
+	s.config = config
+	s.logger = mon.NewLogger()
+
+	s.repo, err = db_repo.New(s.config, s.logger, db_repo.Settings{
+		Metadata: TestModelMetadata,
+	})
+
+	s.NoError(err)
+}
+
+func (s *DbRepoQueryTestSuite) TearDownSuite() {
+	s.mocks.Shutdown()
+}
+
+func (s *DbRepoQueryTestSuite) TestQueryCorrectModel() {
+	ctx := context.Background()
```

Comment:
What's the use case of an integration test here? What do you need the running DB instance for?

Reply:
making sure that the correct query works (when using the correct model)